### PR TITLE
pow: Remove Mul/Div by arbitrary integer types

### LINKED ISSUE #1632: `U265` implements `Mul<T>` for various integer types
> Multiplication by various integers seems to be a bit odd. No other Rust type has it.

In `U265` (private integer type used as inner value of `Target` and `Work`) we implement `Mul` by various integer types, perhaps this was wrong?

### PR DESCRIPTION
**Pull request by tcharding**
When we added `Target` and `Work` types we implemented multiplication and division by anything `Into<u64>`, this is not typically done in the Rust stdlib and also is semantically incorrect for the types.

Remove `Mul` and `Div` impls from `Target` and `Work`. Also remove `Mul<T>` for `T: Into<u64>` from the private `U256` type.

Fix #1632

### CODE REVIEW

**Comment by Kixunil** [comment]
It is semantically correct for `Work`, even though probably useless.

**Review by Kixunil — DISMISSED** [review]
ACK 6503f2e32164d6e80202e8c4a94373142cf84d98

Despite `Mul` making sense for `Work` because it's probably very rarely used if ever. We can add it back later if needed.

File: bitcoin/src/pow.rs [hunk 1]
```diff
@@ -1652,27 +1595,11 @@ mod tests {
     #[test]
     #[should_panic]
     fn u256_overflowing_subtraction_panics() { let _ = U256::ZERO - U256::ONE; }
-
```
Comment (apoelstra) [inline bitcoin/src/pow.rs:1655]: trailing whitespace

Comment (tcharding) [reply bitcoin/src/pow.rs:1655]: oooh, I must have skipper running the githook - my bad. Thanks

Comment (tcharding) [reply bitcoin/src/pow.rs:1655]: The githook misses that whitespace - nooooooo

**Review by Kixunil — APPROVED** [review]
ACK dd316e4d14c0dee0b3e0e705c63fde4dc6d9c304

**Pull request merged**